sphinx_doc_src/fish_sphinxext.py

```diff
@@ -0,0 +1,82 @@
+import os.path
+import re
+
+from docutils import nodes
+from docutils.parsers import rst
+from docutils.statemachine import ViewList
+
+STRING_DOC = os.path.join(os.path.dirname(__file__), "cmds/string.rst")
+with open(STRING_DOC) as fd:
+    STRING_DOC_LINES = list(fd)
+
+
+def string_doc_lines(start, stop):
+    """Print lines from the string documentation start from the line that matches start
```

Comment:
Like this - there is nothing here specific to `string`. It's a function that matches between two regexes (which is also a string command I've been toying with)